### PR TITLE
Issue#530 join organization

### PR DESCRIPTION
**Pull request by Hugo-Hontans**
- [x] new page: organization-find
- [x] create NoOrganizationModule 
- [x] review FeedbackMessage

### CODE REVIEW

File: libs/notification/invitation/+state/invitation.service.ts [hunk 1]
```diff
@@ -15,6 +16,23 @@ export class InvitationService {
     private db: FireQuery
   ) {}
 
+  /** Create an invitation for user asks to join an organization */
+  public sendInvitationToOrg(organization: Partial<Organization>): Promise<void> {
+    const user = this.authQuery.user;
+    const invitation = createInvitation({
+      id: this.db.createId(),
+      message: 'A user would like to participate to your organization',
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.service.ts]: Why it's not in the createInvitation function directly ?

File: libs/notification/invitation/+state/invitation.service.ts [hunk 2]
```diff
@@ -15,6 +16,23 @@ export class InvitationService {
...
+      orgId: organization.id,
+      userInformations: {
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.service.ts]: You should pass a "user" parameter in the createInvitation function to do that.

File: libs/organization/src/lib/pages/organization-find/organization-find.component.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  selector: 'organization-find',
+  templateUrl: './organization-find.component.html',
+  styleUrls: ['./organization-find.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+
+export class OrganizationFindComponent implements OnInit, OnDestroy {
+  public addOrganizationForm: FormGroup;
+  public orgOptions: Organization[];
+  private selectedOrganization: Partial<Organization>;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-find/organization-find.component.ts]: You're already in the organization component. Just "selected" should be fine.

File: libs/organization/src/lib/pages/organization-find/organization-find.component.ts [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    return call({ prefix }).then(matchingOrgs => matchingOrgs.data);
+  }
+
+  private async onChange() {
+    this.addOrganizationForm.valueChanges
+      .pipe(
+        debounceTime(300),
+        distinctUntilChanged(),
+        takeUntil(this.destroyed$)
+      ).subscribe(typingOrgName => {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-find/organization-find.component.ts]: Use async/await here.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html [hunk 5]
```diff
@@ -6,45 +6,34 @@ <h2>Join the community</h2>
...
-          <h4 class="firstTitle">Create your organization</h4>
-          <p>
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec luctus dictum metus quis
-            sagittis.
-          </p>
-        </div>
-        <a routerLink="../create"><mat-icon>keyboard_arrow_right</mat-icon></a>
-      </mat-card-content>
+      <a routerLink="../create">
+        <mat-card-content fxLayout="row" fxLayoutAlign="space-around center">
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.html]: Maybe add a ripple effect if this is a link.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.scss [hunk 6]
```diff
@@ -17,6 +17,11 @@ article {
   background-size: cover;
 }
 
+a {
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.scss]: Don't do it here. Do it at the style.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html [hunk 7]
```diff
@@ -6,45 +6,34 @@ <h2>Join the community</h2>
       sagittis. Duis a auctor nulla, a finibus mauris. Nullam eu maximus lorem.
     </p>
     <mat-card class="mat-elevation-z0">
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.html]: Another card ? I thought we wanted to get read of that as much as possible.

Comment (Hugo-Hontans) [reply libs/organization/src/lib/pages/organization-home/organization-home.component.html]: For this element, I think it's the more simple to use

File: libs/notification/invitation/+state/invitation.service.ts [hunk 8]
```diff
@@ -15,6 +16,17 @@ export class InvitationService {
     private db: FireQuery
   ) {}
 
+  /** Create an invitation for user asks to join an organization */
+  public sendInvitationToOrg(organization: Partial<Organization>): Promise<void> {
+    const user = this.authQuery.user;
+    const message = `${user.name} ${user.surname} would like to participate to your organization`;
+    const invitation = createInvitation(
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.service.ts]: See comment above

File: libs/notification/invitation/+state/invitation.model.ts [hunk 9]
```diff
@@ -1,20 +1,25 @@
...
+  user?: User;
+  orgId: string;
   path: string;
   docInformations: DocInformations;
   state: 'accepted' | 'declined' | 'pending';
   date: Timestamp;
 };
+
+export function createInvitation(params: Partial<Invitation>, user?: User): Invitation {
+  return {
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.model.ts]: Ok, actually the best is to do that : 
```typescript
export function createInvitation(params: Partial<Invitation> = {}): Invitation {
  function createUser(user: ... = {}) {
    return {
      // All default values of user
      ...user,
    }
    return {
      // all default values of invitation
     user: createUser(params.user)
    }
  }
}
```

File: apps/delivery/delivery/src/app/app-routing-module.ts [hunk 10]
```diff
@@ -71,6 +73,14 @@ export const routes: Routes = [
         path: 'organization-home',
         component: OrganizationHomeComponent
       },
+      {
```
Comment (GrandSchtroumpf) [inline apps/delivery/delivery/src/app/app-routing-module.ts]: Should not be directly in the AppModule. We want to put it in the organization lib. Also ideally, we want to lazyload them specificaly (most of the time you don't look for an organization, there is no reason to load thoses components then).

File: libs/organization/src/lib/pages/organization-find/organization-find.component.ts [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+  selector: 'organization-find',
+  templateUrl: './organization-find.component.html',
+  styleUrls: ['./organization-find.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+
+export class OrganizationFindComponent implements OnInit, OnDestroy {
+  public orgControl = new FormControl();
+  public orgOptions: Organization[];
+  private selected: Partial<Organization>;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-find/organization-find.component.ts]: Put private values on top for consistency with others classes.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html [hunk 12]
```diff
@@ -6,45 +6,33 @@ <h2>Join the community</h2>
...
-            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec luctus dictum metus quis
-            sagittis.
-          </p>
-        </div>
-        <!-- TODO: routerLink -->
-        <a><mat-icon>keyboard_arrow_right</mat-icon></a>
-      </mat-card-content>
+      <a routerLink="../organization-find">
+        <mat-card-content matRipple fxLayout="row" fxLayoutAlign="space-around center">
+          <mat-icon svgIcon="magnifying_glass"></mat-icon>
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.html]: Use camelCase for the name of the svgIcon.

File: libs/ui/src/lib/congratulation/congratulation.module.ts [hunk 13]
```diff
@@ -0,0 +1,13 @@
...
+import { FlexLayoutModule } from '@angular/flex-layout';
+import { CongratulationComponent } from './congratulation.component';
+import { RouterModule } from '@angular/router';
+
+@NgModule({
+  declarations: [CongratulationComponent],
+  imports: [CommonModule, MatButtonModule, FlexLayoutModule, RouterModule],
+  exports: [CongratulationComponent],
+})
+export class CongratulationModule { }
```
Comment (GrandSchtroumpf) [inline libs/ui/src/lib/congratulation/congratulation.module.ts]: Maybe use FeebackComponent instead.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html (hunk 12 above)
Comment (Hugo-Hontans) [reply libs/organization/src/lib/pages/organization-home/organization-home.component.html]: I create the issue #674 for that

File: apps/delivery/delivery/src/app/app-routing-module.ts (hunk 10 above)
Comment (Hugo-Hontans) [reply apps/delivery/delivery/src/app/app-routing-module.ts]: So I created NoOrganizationModule for that

File: libs/notification/invitation/+state/invitation.model.ts [hunk 14]
```diff
@@ -1,20 +1,28 @@
...
   path: string;
   docInformations: DocInformations;
   state: 'accepted' | 'declined' | 'pending';
   date: Timestamp;
-};
+}
+
+export function createInvitation(params: Partial<Invitation> = {}): Invitation {
+  function createUser(user: Partial<User> = {}) {
+    return { ...user };
```
Comment (GrandSchtroumpf) [inline libs/notification/invitation/+state/invitation.model.ts]: If there is no default params or if you want the whole user you don't need that ...
But if I remember well you only wanted some fields of the user.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html (hunk 12 above)
Comment (GrandSchtroumpf) [reply libs/organization/src/lib/pages/organization-home/organization-home.component.html]: You could reference it here. Like that when you to it, you can search for the issue more easily

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html (hunk 7 above)
Comment (GrandSchtroumpf) [reply libs/organization/src/lib/pages/organization-home/organization-home.component.html]: Nope. You need a Navigation List : 
```html
<mat-nav-list class="mat-elevation-z1">
  <a routerLink="../create">
    <mat-icon mat-list-icon svgIcon="adjustable_wrench"></mat-icon>
    <h4 mat-line>Create your organization</h4>
    <p mat-line>...</p>
  </a>
  <a>...</a>
</mat-nav-list>
```
And add `mat-nav-list { border-radius: 4px; }` in this component or in the theme (I think that this is everywhere)

File: apps/delivery/delivery/src/app/app-routing-module.ts [hunk 15]
```diff
@@ -29,6 +28,10 @@ export const routes: Routes = [
         redirectTo: 'o',
         pathMatch: 'full'
       },
+      {
+        path: 'create-organization',
```
Comment (GrandSchtroumpf) [inline apps/delivery/delivery/src/app/app-routing-module.ts]: Call it "org" or "organization". Avoid the "-" in names of path.

File: libs/organization/src/lib/pages/organization-home/organization-home.component.html [hunk 16]
```diff
@@ -5,35 +5,33 @@ <h2>Join the community</h2>
...
+    <mat-nav-list>
       <a routerLink="../create">
-        <mat-card-content matRipple fxLayout="row" fxLayoutAlign="space-around center">
-          <mat-icon svgIcon="adjustable_wrench"></mat-icon>
-          <div class="card-text">
-            <h4 class="firstTitle">Create your organization</h4>
-            <p>
+        <mat-list-item>
+          <mat-icon matListIcon svgIcon="adjustable_wrench"></mat-icon>
+          <div class="item-text">
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.html]: Remove this div (see comment on css below)

File: libs/organization/src/lib/pages/organization-home/organization-home.component.scss [hunk 17]
```diff
@@ -1,14 +1,27 @@
...
-  background-color: var(--whiteBackground);
   height: 100%;
 }
 
 article {
   width: 500px;
 }
 
+.item-text {
+  padding-left: 40px;
```
Comment (GrandSchtroumpf) [inline libs/organization/src/lib/pages/organization-home/organization-home.component.scss]: Set padding on h4 and p directly.

File: styles/components/mat-list.scss [hunk 18]
```diff
@@ -0,0 +1,9 @@
+.mat-nav-list {
+  border-radius: 10px;
+  border: 1px solid var(--borderGrey);
+}
+
+.mat-list-base .mat-list-item .mat-line {
+  white-space: normal !important;
```
Comment (GrandSchtroumpf) [inline styles/components/mat-list.scss]: I don't understand why... It will break the principle of mat-line...

**Pull request merged**